recipes/libcusparse/meta.yaml

```
@@ -0,0 +1,156 @@
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: libcusparse
+    files:
+      - lib/libcusparse.so.*                             # [linux]
+      - targets/{{ target_name }}/lib/libcusparse*.so.*  # [linux]
+      - Library\bin\cusparse*.dll                        # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+        - libnvjitlink
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - {{ pin_compatible("libnvjitlink", max_pin="x.x") }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      commands:
+        - test -L $PREFIX/lib/libcusparse.so.{{ version }}                                  # [linux]
+        - test -L $PREFIX/lib/libcusparse.so.{{ version.split(".")[0] }}                    # [linux]
+        - test -L $PREFIX/{{ target_name }}/lib/libcusparse.so.{{ version }}                # [linux]
+        - test -f $PREFIX/{{ target_name }}/lib/libcusparse.so.{{ version.split(".")[0] }}  # [linux]
+        - if not exist %LIBRARY_BIN%\\cusparse*.dll exit 1                                  # [win]
+    about:
+      home: https://developer.nvidia.com/cusparse
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: CUDA Sparse Matrix Library
+      description: |
+        cuSPARSE - Basic Linear Algebra for Sparse Matrices on NVIDIA GPUs
+      doc_url: https://docs.nvidia.com/cuda/cusparse/index.html
+
+  - name: libcusparse-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libcusparse", max_pin="x") }}
+    files:
+      - lib/libcusparse.so         # [linux]
+      - lib/pkgconfig                                 # [linux]
+      - targets/{{ target_name }}/include             # [linux]
+      - targets/{{ target_name }}/lib/libcusparse.so  # [linux]
+      - targets/{{ target_name }}/lib/stubs           # [linux]
+      # Not shipping Fortran bindings (in src) on Windows for now
+      - targets/{{ target_name }}/libcusparse/src     # [linux]
+      - Library\include                             # [win]
+      - Library\lib                                 # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - {{ pin_subpackage("libcusparse", exact=True) }}
+      run_constrained:
+        - libcusparse-static >={{ version }}  # [linux]
```

Add arm-variant constraint.
```suggestion
        - arm-variant * {{ arm_variant_type }} # [aarch64]
        - libcusparse-static >={{ version }}  # [linux]
```

---

Technically we don't need `arm-variant` for `-dev` packages, as it'd get this requirement from the `lib` package. But I am fine following the same style.

---

Agreed -- but let's match the style. There are potentially other factors that would make this matter, rather than being only a transitive dependency/constraint. (At one point we had discussed the possibility of `-dev` packages shipping only headers and stubs, which would eliminate the dependency on the `lib` package.)